lib/tasks/unpublish.rake

```diff
@@ -0,0 +1,42 @@
+require "services"
+
+namespace :unpublish do
+  desc "Unpublish a Finder and redirect the finder page to specific URL"
+  task :redirect_finder, %i[finder_slug redirect_url] => :environment do |_, args|
+    schema_file = Dir.glob("lib/documents/schemas/*.json").find do |file|
+      File.foreach(file).grep(/"base_path": "\/#{args.finder_slug}"/).any?
+    end
+
+    if schema_file.nil?
+      puts "Could not find any finders with that slug. Please check again."
+      next
+    end
+
+    schema = MultiJson.load(File.read(schema_file))
+    puts "=== Finder found ==="
+    puts "Slug: #{args.finder_slug}"
+    puts "Finder name: #{schema['name']}"
+    puts "Content ID: #{schema['content_id']}"
+    puts "Redirecting to: #{args.redirect_url}"
+    puts "Are you sure you want to unpublish this finder? (y/n)"
+
+    user_approval = $stdin.gets.strip.downcase
+    if user_approval != "y"
+      puts "Cancelled action. Not unpublishing anything."
+      next
+    end
```

Comment:
As we discussed in Slack: I don't think we gain user consent before performing the unpublish action.

Rake tasks haven't traditionally run in interactive TTYs in GOV.UK, so the convention is for Rake tasks to 'just do the thing' rather than wait for user input.

If a developer misfires and accidentally unpublishes the wrong thing, they always have the option of republishing it with the `publishing_api:publish_finder` task.

Reply:
Sure i can remove